finetuner/tuner/pytorch/losses.py

```diff
@@ -174,3 +174,46 @@ def get_default_miner(
             return TripletMiner()
         else:
             return TripletSessionMiner()
+
+
+class NTXentLoss(nn.Module):
+    """Compute the NTXent (Normalized Temeprature Cross-Entropy) loss.
+
+    This loss function is a temperature-adjusted cross-entropy loss, as defined in the
+    `SimCLR paper <https://arxiv.org/abs/2002.05709>`. It operates on batches where
+    there are two versions of each instance (label)
+    """
+
+    def __init__(self, temperature: float = 0.5) -> None:
+        """Initialize the loss instance.
+
+        :param temerature: The temperature parameter
+        """
+        super().__init__()
+
+        self.temperature = temperature
+
+    def forward(self, embeddings: torch.Tensor, labels: torch.Tensor) -> torch.Tensor:
```

Comment:
It may even be much easier to reason about this loss if the interface shows `left_embeddings` and `right_embeddings` so that u know that by `zipping` them u can find the positive correspondances